ocaml/xapi/xapi_pool.ml

```diff
@@ -3717,3 +3754,52 @@ let get_guest_secureboot_readiness ~__context ~self:_ =
       `ready_no_dbx
   | _, _, _, _ ->
       `not_ready
+
+let put_bundle_handler (req : Request.t) s _ =
+  req.Request.close <- true ;
+  Xapi_http.with_context "Sync bundle" req s (fun __context ->
+      (* This is the signal to say we've taken responsibility from the CLI server
+         for completing the task *)
+      (* The GUI can deal with this itself, but the CLI is complicated by the thin
+         cli/cli server split *)
+      TaskHelper.set_progress ~__context 0.0 ;
+      Pool_features.assert_enabled ~__context ~f:Features.Updates ;
+      let pool = Helpers.get_pool ~__context in
+      Xapi_pool_helpers.with_pool_operation ~__context ~self:pool
+        ~doc:"pool.sync_bundle" ~op:`sync_bundle
+      @@ fun () ->
+      Http_svr.headers s (Http.http_200_ok ()) ;
+      let repo =
+        Repository_helpers.get_single_enabled_update_repository ~__context
+      in
+      match Db.Repository.get_origin ~__context ~self:repo with
+      | `bundle -> (
+          let result =
+            Tar_ext.unpack_tar_file
+              ~dir:!Xapi_globs.bundle_repository_dir
+              ~ifd:s
+              ~max_size_limit:!Xapi_globs.bundle_max_size_limit
+          in
+          match result with
+          | Ok () ->
+              TaskHelper.set_progress ~__context 0.8 ;
```

Comment:
Do we need to set the progress to 1.0 finally?

Reply:
No. 1.0 will be set automatically when the task succeeds/fails/cancels in `TaskHelper` (`Db_actions.DB_Action.Task.set_progress ~__context ~self ~value:1.` ).